### PR TITLE
only patch the event loop policy on old versions of tornado

### LINKED ISSUE #12018: support the proactor eventloop on windows with tornado >=6.1
### Problem description

the SelectorEventLoop is forced on python 3.8+ for windows, which has a number of disadvantages - eg `asyncio.create_subprocess_exec` is not supported on the SelectorEventLoop.

### Feature description

tornado 6.1 added support for running on the ProactoEventLoop by running `select` in a thread: https://github.com/tornadoweb/tornado/issues/2608#issuecomment-736185273

### Potential alternatives

always support the ProactorEventLoop - and set the minumum tornado version to 6.1

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by graingert**
- [x] issues: fixes #12018
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by bryevdv** [comment]
@graingert Thanks for the PR. I think ideally we would update one of the test envs to install tornado 5.1 and then we would leave the fixup tests in. It's a little to soon to drop Tornado 5.x support but we should make sure testing covers it. Alternatively we could make a new env file for the `minimal-deps` test that specifies Tornado 5.1. Currently the `minimal-deps` test just uses an existing CI file and removes some optional packages after the fact. But it would be fine/good to have a dedicated env file for that case. Do you have some bandwidth to keep pushing on this PR a bit more?

**Comment by graingert** [comment]
@bryevdv tornado 5.1 doesn't support python 3.8+

**Comment by bryevdv** [comment]
@graingert Right, from the tornado docs

> Tornado 5.x runs on Python 2.7, and 3.4+

And AFAIK also 5.x does in fact work Windows with the event loop fixup (modulo the disadvantages of using `SelectorEventLoop`). I am all for encouraging users to update their Tornado version ASAP, but some users have upgrade constraints to contend with, so I think it would be better to drop Tornado 5.x support a little further in to 2022.

**Comment by graingert** [comment]
@bryevdv ok I've pushed that now - but the minimal-deps build seems to have got stuck waiting for a runner to pick it up " Started 2h 3m 43s ago "

**Comment by graingert** [comment]
oh woops that was my mistake I didn't use a `strategy.matrix`

**Comment by bryevdv** [comment]
@graingert sorry I got distracted away by some other things, how would you like to proceed on this? The failing tests report:
```
bokeh/client/connection.py:37: in <module>
    from tornado.httpclient import HTTPClientError, HTTPRequest
E   ImportError: cannot import name 'HTTPClientError' from 'tornado.httpclient' (/usr/share/miniconda3/envs/bk-test/lib/python3.8/site-
```
But the [Tornado documentation](https://www.tornadoweb.org/en/stable/httpclient.html#tornado.httpclient.HTTPClientError) states that import should be present in 5.1: 

> Changed in version 5.1: Renamed from HTTPError to HTTPClientError

File: ci/environment-test-minimal-deps.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  - python=3.8
+
+  # runtime
+  - jinja2 >=2.7
+  - numpy >=1.11.3
+  - packaging >=16.8
+  - pillow >=4.0
+  - python-dateutil >=2.1
+  - pyyaml >=3.10
+  - tornado <=5.1
```
Comment (graingert) [inline ci/environment-test-minimal-deps.yml:14]:
```suggestion
  - tornado >5.1, <6
```

Comment (bryevdv) [inline ci/environment-test-minimal-deps.yml:14]: @graingert perhaps this needs to be more strict e.g. `tornado 5.1*` ?

Comment (graingert) [reply ci/environment-test-minimal-deps.yml:14]: I'm actually not sure on the syntax for conda environments

File: ci/environment-test-minimal-deps.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  - python=3.8
+
+  # runtime
+  - jinja2 >=2.7
+  - numpy >=1.11.3
+  - packaging >=16.8
+  - pillow >=4.0
+  - python-dateutil >=2.1
+  - pyyaml >=3.10
+  - tornado >5.1, <6
```
Comment (graingert) [inline ci/environment-test-minimal-deps.yml:14]:
```suggestion
  - tornado 5.1.*
```

**Comment by mattpap** [comment]
@graingert, thanks.

**Pull request merged**